ci/infra/testrunner/tests/test_istio_deployment.py

```diff
@@ -0,0 +1,111 @@
+        prefix: /status
+    - uri:
+        prefix: /delay
+    route:
+    - destination:
+        port:
+          number: 8000
+        host: httpbin
+EOF
+""")
+
+
+@pytest.mark.pr
+def test_istio_deployment(deployment, platform, skuba, kubectl):
+    logger = logging.getLogger("testrunner")
+    logger.info("Deploy Istio")
+    istioctl = ("""
+                istioctl --kubeconfig={config} manifest apply \
+                        --set profile=default \
+                        --set addonComponents.prometheus.enabled=false \
+                        --set hub=registry.suse.de/devel/caasp/5/containers/containers/caasp/v5 \
+                        --set tag=1.5.4 \
+                        --set values.pilot.image=istio-pilot \
+                        --set values.global.proxy.image=istio-proxyv2 \
+                        --set values.global.proxy_init.image=istio-proxyv2
+                 """.format(config=kubectl.get_kubeconfig()))
+
+    kubectl.utils.runshellcommand(istioctl)
+
+    kubectl.run_kubectl("-n istio-system wait --for=condition=available deploy/istio-ingressgateway --timeout=3m")
+
+    logger.info("Deploy httpbin sample app")
+    kubectl.run_kubectl("create -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/httpbin/httpbin.yaml")
+
+    logger.info("Create the istio config")
+    kubectl.run_kubectl("apply -f - << EOF " + GATEWAY_HTTPBIN)
+    kubectl.run_kubectl("apply -f - << EOF " + VIRTUALSERVICE_HTTPBIN)
+
+    # Wait for istio to digest the config
+    time.sleep(100)
+
+    nodePort = kubectl.run_kubectl("-n istio-system get service/istio-ingressgateway -o jsonpath='{ .spec.ports[1].nodePort }'")
+
+    wrk_idx = 0
+    ip_addresses = platform.get_nodes_ipaddrs("worker")
+
+    url = "{protocol}://{ip}:{port}{path}".format(protocol="http", ip=str(ip_addresses[wrk_idx]), port=str(nodePort), path="/status/200")
+    r = requests.get(url, headers={'host': 'httpbin.example.com'})
+
+    assert 200 == r.status_code
+
+    # Cleanup
+    kubectl.run_kubectl("delete -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/httpbin/httpbin.yaml")
+    istioctl_delete = ("""
+                istioctl --kubeconfig={config} manifest generate \
+                        --set profile=default \
+                        --set addonComponents.prometheus.enabled=false \
+                        --set hub=registry.suse.de/devel/caasp/5/containers/containers/caasp/v5 \
+                        --set tag=1.5.4 \
+                        --set values.pilot.image=istio-pilot \
+                        --set values.global.proxy.image=istio-proxyv2 \
+                        --set values.global.proxy_init.image=istio-proxyv2 \
+                        | kubectl --kubeconfig=/home/manuel/manuelbuil/testrunner/test-cluster/admin.conf delete -f - || true
+                 """.format(config=kubectl.get_kubeconfig()))
+    kubectl.utils.runshellcommand(istioctl_delete)
+
+    with pytest.raises(Exception):
```

Comment:
Not sure about the objective of this test. Do you want to verify istio has been removed?

Reply:
If there is an error, I'd like to check if all istio stuff exists and how does the service look like

Reply:
I have my doubts here. Looks like you are trying to debug the test. This looks like not a good practice, but maybe I'm biased towards simple pass/don't pass tests.

Reply:
You are right, I was confusing the usage of pytest.raises